synapse/replication/http/_base.py

```diff
@@ -243,6 +271,20 @@ def register(self, http_server):
             handler = self._cached_handler  # type: ignore
             url_args.append("txn_id")
 
+        # If there's a shared replication secret, wrap the handler to ensure the
+        # secret is checked.
+        if self._replication_secret:
+            raw_handler = handler
+
+            def check_auth_and_handle(request, *args, **kwargs):
+                """
+                Check the authorization headers, then handle the request.
+                """
+                self.check_auth(request)
+                return raw_handler(request, *args, **kwargs)
+
+            handler = check_auth_and_handle
```

Comment:
I think its probably cleaner to always wrap the function in `_cached_handler` and put `self.check_auth` in there too (and renaming it)

Reply:
Are there replication endpoints that don't have `CACHE` set to `True`? Or do you think we should move that check into `_cached_handler`?

Reply:
The latter is what I was thinking

Reply:
Done!